types/felt_test.go

```diff
@@ -0,0 +1,106 @@
+package types
+
+import (
+	"fmt"
+	"bytes"
+	"encoding/json"
+	"math/big"
+	"strconv"
+	"testing"
+)
+
+var (
+	rawTest string = `{
+		"max_felt": 3618502788666131213697322783095070105623107215331596699973092056135872020481,
+		"felts": [
+			{"value": "0x0", "expected": 0},
+			{"value": "0x1277312773", "expected": 79309121395},
+			{"value": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB", "expected": "1071767867375995473349368877325274214414350531515"},
+			{"value": "2458502865976494910213617956670505342647705497324144349552978333078363662855", "expected": "2458502865976494910213617956670505342647705497324144349552978333078363662855"}
```

Comment:
You could add shorts as part of the test and test that it will fail if shorts are longer than 31 characters or includes non supported characters.